### ISSUE TITLE
mozilla.org

### ISSUE THREAD

**Issue by CharliesChocolateFactory** [uBlockOrigin/uAssets#8114]
### URL(s) where the issue occurs

https://addons.mozilla.org/firefox/

### Describe the issue

There is sponsored content that is not hidden, the "Sponsored extensions" space.

### Versions

- Browser/version: Firefox 82.0.2
- uBlock Origin version: 1.30.6

### Settings

Default.

### Notes

Ad removal seems to be denied by Easylist:

https://forums.lanik.us/viewtopic.php?f=62&t=45271

I think that this should be removed by default just like Amazon sponsored content.

**Comment by krystian3w** [comment]
No works without modified in `about:config`:

`extensions.webextensions.restrictedDomains`
`privacy.resistFingerprinting.block_mozAddonManager`

Source:
https://github.com/openstyles/stylus/blob/c416fa7ca0929012805f4df04959e48486f73bd4/_locales/en/messages.json#L1444
https://github.com/openstyles/stylus/blob/c416fa7ca0929012805f4df04959e48486f73bd4/_locales/en/messages.json#L1448

**Comment by uBlock-user** [comment]
https://github.com/gorhill/uBlock/wiki/Privileged-Pages

**Comment by CharliesChocolateFactory** [comment]
The ad removal can be effective with the right Firefox settings, or if for example a user of another browser with ublock origin visits those pages. There is no reason to deny the ad removal for those whose browsers/settings allow ad removal to happen. It's also a matter of principle:

> Adverts must be blocked regardless of the reasons for their presence or the purpose that the money raised ultimately serves.

Why close the bug so fast without even allowing a discussion ? Clearly there is some controversy here.

**Comment by uBlock-user** [comment]
> Why close the bug so fast without even allowing a discussion ? 

You just responded, so discussion is allowed, the problem you cannot that site and all the ones listed in https://github.com/gorhill/uBlock/wiki/Privileged-Pages by default and since they cannot be filtered by default, adding filters will not help uBo users either. 

You can do what was suggested in https://github.com/gorhill/uBlock/wiki/Privileged-Pages and add the filters then for yourself.

**Comment by uBlock-user** [comment]
> Why close the bug so fast without even allowing a discussion ? 

You just responded, so discussion is allowed, the problem is users cannot filter that site and all the ones listed in https://github.com/gorhill/uBlock/wiki/Privileged-Pages by default and since they cannot be filtered by default, adding filters will not help uBo users either. 

You can do what was suggested in https://github.com/gorhill/uBlock/wiki/Privileged-Pages and add the filters then for yourself.

**Comment by CharliesChocolateFactory** [comment]
> You just responded, so discussion is allowed

Yes but the bug was already closed with a "wontfix" tag.

> since they cannot be filtered by default, adding filters will not help uBo users either.

As I said, it will help the uBo users whose browsers/settings allow ad removal to happen.

If Google decided to make Chrome disable adblocking on Google sites unless one changes some browser settings to re-enable adblocking, would you say that uBo should no longer even attempt to add the needed filters ?

**Comment by uBlock-user** [comment]
> it will help the uBo users whose browsers/settings allow ad removal to happen.

Those users and you can add the filter yourself `addons.mozilla.org##.Card--no-footer.Card--photon.AddonsCard--horizontal.SponsoredAddonsShelf.AddonsCard.CardList.Card`

This is a self-ad, hypothetically, if it were to be added, it would go into annoyances list(which is not enabled by default) and not into the main list as you're expecting.

> If Google decided to make Chrome disable adblocking on Google sites

Google already does on `https://chrome.google.com/webstore/category/extensions`.

**Comment by CharliesChocolateFactory** [comment]
> Those users and you can add the filter yourself

Why should those whose browsers/settings allow ad removal not benefit from default blocking for them at least, and should add it manually ? See again my Google example.

> Google already does

And are ad filters not added, or would they be not added, there by uBo only for the reason that Google does this ?

> This is a self-ad, hypothetically, if it were to be added, it would go into annoyances list(which is not enabled by default) and not into the main list as you're expecting.

I think that there is disagreement on this being only a self-ad that should not be blocked by default. According to another contributor [in the Easylist forum thread](https://forums.lanik.us/viewtopic.php?p=157082#p157082) that I started:

> "Sponsored" is an ad, not a self promotion. Amazon use the same trick. Is it "legal" for adblocking rules?

**Comment by LanikSJ** [comment]
> Why should those whose browsers/settings allow ad removal not benefit from default blocking for them at least, and should add it manually ? See again my Google example.

Not everyone wants those "Sponsored Ads" blocked some people want to see them, especially if they aren't too obnoxious. 

> I think that there is disagreement on this being only a self-ad that should not be blocked by default. According to another contributor [in the Easylist forum thread](https://forums.lanik.us/viewtopic.php?p=157082#p157082) that I started:
> 
> > "Sponsored" is an ad, not a self promotion. Amazon use the same trick. Is it "legal" for adblocking rules?

The poster you've quoted isn't an author of any filters lists that I'm aware off so sadly his/her opinion just doesn't matter. 

I have my own filter lists I maintain for this specific reason if I don't like something I add it.

**Comment by Yuki2718** [comment]
Note: I explicitly stated:
> Though I'm not in a responsible place to answer

And what I said are align with @uBlock-user and @LanikSJ . The quoted comment is not mine.

**Comment by CharliesChocolateFactory** [comment]
> The poster you've quoted isn't an author of any filters lists that I'm aware off so sadly his/her opinion just doesn't matter.

He is a big Liste FR contributor, and he is just repeating what the Liste FR policy seems to be is according to the Liste FR author too, as is discussed in the Easylist forum thread. I pointed out what seemed to me like an inconsistency between the two policies, but these arguments have been ignored there and here. I think that the Liste FR author policy is more in line with the values of ublock origin than the Easylist author policy and this is what I wanted to discuss here for ublock origin specific filters. I have noted other policy differences in the past, for example Liste FR blocking coin miners by default (just like ublock origin's own filters) while Easylist preferred to relegate this to the EasyPrivacy list that is not enabled by default in some blockers. I remember too the Liste FR author objecting for privacy reasons to the Easylist forum adding social media logins, that were then removed, which seems in line with the ublock origin values too. I don't know who is the ublock origin specific list author who decides here, is it uBlock-user ?

> Not everyone wants those "Sponsored Ads" blocked some people want to see them, especially if they aren't too obnoxious.

What kind of adblocker user would want to see those sponsored addons ? They are only listed prominently because they paid for it, they have zero relevance to the site visitor beyond that, the page has unpaid recommendations just below. It could even be argued that the traditional third-party ads can be more relevant than that to the visitor needs, and are still rightfully blocked... The only reason I could imagine is whitelisting the ad by solidarity with the site, but I don't think that it's the philosophy of ublock origin to do that by default. And once we accepted this for Mozilla, it will be harder to reject this for similar ads on other sites like Amazon that are apparently only blocked currently by Easylist by accident, not by policy.

> AG also has allowlist for search ads enabled by default, and we got a FP report which they couldn't get the destination due to a DNS rule blocking an ad-server when the reporter clicked search ads. What I wanna say is people need to understand how versatile users' needs are, and it's simply impossible to meet all of them by single list.

I don't know if the AdGuard company you contribute to and the Brave employer of the Easylist author whitelist Google ads in their own content blocker product only because some people want to see them or if there are other reasons. But I am glad that ublock origin does still block them by default. For how long ?

**Comment by uBlock-user** [comment]
> I don't know who is the ublock origin specific list author who decides here, is it uBlock-user ?

uBO-team, consisting of few volunteers and gorhill and as per internal discussion with **gorhill** it has been agreed upon to follow EL's decision.

**Comment by CharliesChocolateFactory** [comment]
> uBO-team, consisting of few volunteers and gorhill and as per internal discussion with gorhill it has been agreed upon to follow EL's decision.

That was fast. Where can this internal discussion be read please ?

**Comment by mapx-** [comment]
internal team discussion, not public

**Comment by okiehsch** [comment]
> Where can this internal discussion be read please ?

You can't that's why it was "internal" only members can read it.

It was not some elaborate discussion, basically the posed question was 
" Should we add filters for pages listed in Privileged-Pages for Firefox ?"

And everybody agreed that we should go along with whate EasyList decides.

**Comment by CharliesChocolateFactory** [comment]
> You can't that's why it was "internal" only members can read it.

Why do that ?

**Comment by uBlock-user** [comment]
> Where can this internal discussion be read please ?

Only visible to team members -- https://user-images.githubusercontent.com/21290713/97776626-aeb06080-1b8f-11eb-9a64-aa709daa7294.PNG

The case for rejection here on uBO is quite clear --

1) Adding filters for a site where uBO and any other extension is intentionally tasked to not work, is pointless and as for the argument of  "but that will help some Firefox users with custom configs where they can bypass Firefox's restriction" -- we don't add filters for users with any custom Firefox configuration and never have till date and FYI, bypassing such restrictions is simply not possible from within the browse with Chrome/Chromium.

2) As said before this a selfpromo/ad and because of that, if this was to be added, it would have gone to Annoyances list and by rule we're only adding filters to Annoyances list where uBO's extended syntax is required(not for this case as a simple cosmetic filter blocks the self-ad/promo).

**Comment by okiehsch** [comment]
>Why do that ?

I don't follow, you mean we should have no internal discussions?

**Comment by CharliesChocolateFactory** [comment]
> we don't add filters for users with any custom Firefox configuration and never have till date

What about Chrome users who browse AMO ? Is Chrome unsupported too ?

Besides, here the "custom configuration" is allowing the users to bypass a browser system enforcing ads on them on the site of the browser developers. I don't know what previous cases of "custom Firefox configuration" were denied having filters added, but here it would have seemed needed to make a point against such an abusive practice from the browser developers.

> As said before this a selfpromo/ad and because of that, if this was to be added, it would have gone to Annoyances lis

So you're taking the Easylist point of view, not the Liste FR one, if I understand well.

**Comment by CharliesChocolateFactory** [comment]
> I don't follow, you mean we should have no internal discussions?

Why should discussions deciding what should be blocked be hidden ? And what is so funny about asking this ?

**Comment by uBlock-user** [comment]
> Why should discussions deciding what should be blocked be hidden ?

Decision was already made, I asked gorhill only because I wanted to know if he had a different thought process for this.
